### PR TITLE
For #5784: Fix hiding default browser preference in sdk<=23

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -96,9 +97,9 @@ class SettingsFragment : PreferenceFragmentCompat(), AccountObserver {
         preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener(preferenceChangeListener)
 
         if (SDK_INT <= Build.VERSION_CODES.M) {
-            findPreference<DefaultBrowserPreference>(getPreferenceKey(pref_key_make_default_browser))?.apply {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: I think all we need to do is replace `DefaultBrowserPreference` with `Preference`? It looks like we just replaced this in `preferences.xml` and forgot to update it here

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: AFAIK `Preference` does not have an `isVisible` property or a method to hide it. That is why I used `removePreference` from it's parent category. I will check on Monday though if you prefer that way.

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: I'm pretty sure that will work. I may just push a quick patch to this PR so we don't have crashes over the weekend, if you don't mind!

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt]: It's fine by me! :)

**Pull request merged**